servers/zms/src/main/java/com/yahoo/athenz/zms/store/impl/jdbc/JDBCConnection.java

```diff
@@ -5746,4 +5755,92 @@ Boolean nullIfDefaultValue(boolean flag, boolean defaultValue) {
     Integer nullIfDefaultValue(int value, int defaultValue) {
         return value == defaultValue ? null : value;
     }
+    
+    private void addTagsToRoles(Map<String, Role> roleMap, String domainName) {
```

Comment:
This could be a very expensive operation. If a domain has 100 roles (which is not that uncommon), we're going to make 100 sql commands - with just regular latency this will add up quick considerably.

Any reason why we're not just running a query to get all tags for a domain and then parse the results and update accordingly. The roles are already in a map so we can quickly update each role accordingly.

Reply:
This is correct for getRoles.
But this method is still needed for getRole..